src/compiler/diagnosticMessages.json

```diff
@@ -4003,6 +4003,10 @@
         "category": "Message",
         "code": 6220
     },
+    "Disable using source of project reference redirect files.": {

```

Comment:
It’s hard to find the right balance between accuracy and human readability, but I’m not sure most people know what “project reference redirect files” are. Maybe _I_ don’t even know what they are—a redirect file is a declaration file that stands in for a `.ts` source file, right? Would the message

> Use source files instead of declaration files for editor services in referenced projects.

be accurate?

Reply:
`Use source files instead of declaration files from referenced projects` Its in the project referencing other projects and seems like word editor shouldn't be part of the diagnostic message

Reply:
cc @DanielRosenwasser To comment on message to use here

Reply:
I like your recent suggestion Sheetal.

Alternatively,

> Prefer reading source files from referenced projects instead of their declaration files.

Reply:
> Use source files instead of declaration files from referenced projects

sounds pretty good to me. I think it still begs the question “...use for what?” for me, but it’s a bit difficult to explain concisely.